config/routes.rb

```diff
@@ -3,5 +3,7 @@
   resources :your_contributions, only: %i[new create]
   resources :your_results, only: %i[index]
 
+  post 'your_results', to: 'your_results#index'
```

Comment:
Can you remove this line and makes the form uses GET /your_results ? Then you don't need this line and uses the existing route. I am just giving this suggestion because something does not sounds right ... to have a post and get action for the same controller method. Wdyt?